src/Chainweb/Mempool/InMem.hs

```diff
@@ -245,15 +256,16 @@ data InMemConfig t = InMemConfig {
 
 
 ------------------------------------------------------------------------------
-data InMemoryMempool t = InMemoryMempool {
+data InMemoryMempool t cas = InMemoryMempool {
     _inmemCfg :: InMemConfig t
   , _inmemDataLock :: MVar (InMemoryMempoolData t)
   , _inmemBroadcaster :: TxBroadcaster t
   , _inmemReaper :: ThreadId
+  , _inmemBlockHeaderDb :: BlockHeaderDb
```

Comment:
This stuff doesn't belong in here? This data is only needed by the mempool consensus layer, and so you should stuff it in there. Adding these functions here pollutes the api, when you could have

```
data MempoolConsensus t = MempoolConsensus {
    _mpcMempool :: MempoolBackend t,
    _mpcLastNewBlockParent :: Maybe (IORef BlockHeader),
    ...
}
```

Nothing that the consensus fork processing algorithm needs to do should require knowledge of mempool internals beyond that which is published by the api, so it's better to wrap it.